api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionsUriHelper.java

```diff
@@ -0,0 +1,97 @@
+package org.zalando.nakadi.service;
+
+import org.springframework.web.util.UriComponentsBuilder;
+import org.zalando.nakadi.domain.PaginationLinks;
+import org.zalando.nakadi.domain.Subscription;
+import org.zalando.nakadi.repository.db.SubscriptionDbRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class SubscriptionsUriHelper {
+
+    public static PaginationLinks createSubscriptionPaginationLinks(
```

Comment:
I prefer this functionality to be placed in `core-metastore` module along with `SubscriptionDbRepository`.
The reason being the logic in here and the token based query generation in `SubscriptionDbRepository` are working together to provide pagination.

Reply:
Also this function is too big to understand in one go.
It will be nice we we can extract `buildPrevLink` and `buildNextLink`.

Reply:
Changed it back. 
But I would not move the code back, as it is totally wrong. Pagination link is part of view logic, and should not be located in `core-something` project.